cpp/src/arrow/c/bridge.cc

```diff
@@ -1275,7 +1398,19 @@ class ImportedBuffer : public Buffer {
 
 struct ArrayImporter {
   explicit ArrayImporter(const std::shared_ptr<DataType>& type)
-      : type_(type), zero_size_buffer_(std::make_shared<Buffer>(kZeroSizeArea, 0)) {}
+      : type_(type),
+        zero_size_buffer_(std::make_shared<Buffer>(kZeroSizeArea, 0)),
+        device_type_(DeviceType::CPU) {}
+
+  Status Import(struct ArrowDeviceArray* src, const DeviceMemoryMgr& mapper) {
+    ARROW_ASSIGN_OR_RAISE(memory_mgr_,
+                          mapper.get_manager(src->device_type, src->device_id));
+    device_type_ = static_cast<DeviceType>(src->device_type);
+    RETURN_NOT_OK(Import(&src->array));
+    import_->sync_event_ = src->sync_event;
+    memory_mgr_.reset();
```

Comment:
Should `device_type_` be restored as well?

Reply:
If not, I would initialize it to `UNKNOWN` in the ctor and set it to `UNKNOWN` here as `device_type_` doesn't seem to be used when importing from CPU arrays.

Reply:
That's fair, prevents potential issues. I'll set it to `UNKNOWN`

Reply:
Or rather, I'll set it to CPU here afterwards just for consistency in case it ends up eventually getting used for CPU arrays.